### ISSUE TITLE
Error when deployed on OCP Cluster

### ISSUE THREAD

**Issue by cvishal** [FudanSELab/train-ticket#143]
### Summary
ts-ui-dashboard pod has CrashLoopBackOff error, when deployed on OCP Cluster.

### Expected behaviour

All PODs should be running state. I see one pod is in CrashLoopBackOff. Rest all are fine and in running state.

### Current behaviour

When deployed application, I see all pods running fine, except following:
```
oc logs ts-ui-dashboard-5cb87645cb-7gj8d
2021/04/06 01:28:53 [emerg] 1#1: host not found in upstream "ts-avatar-service" in /usr/local/openresty/nginx/conf/nginx.conf:386
nginx: [emerg] host not found in upstream "ts-avatar-service" in /usr/local/openresty/nginx/conf/nginx.conf:386
```

### Steps to reproduce 

Clone this train-ticket repo.
Deploy the application on OCP, using Kubernetes yml files located in 'train-ticket/deployment/kubernetes-manifests/quickstart-k8s' folder.

### Your environment

OS(e.g: `cat /etc/os-release`): OCP 4.3.13

docker version(e.g `docker version`): 
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.2+853223d", GitCommit:"853223d", GitTreeState:"clean", BuildDate:"2020-10-14T15:02:59Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}

 
docker-compose version (e.g `docker-compose --version`):

 
Kubernetes version(use `kubectl version`):

istio version: 

### Additional context

I didn't find anything related to ts-avatar-service.
It doesn't seems to be valid service. 
Or I might be missing something.

**Comment by kagaya85** [comment]
Thanks for bug report, this problem may be caused by a new service we added but the nginx configuration file is not updated in time. 

We're sorry about this and will fix it soon.

**Comment by cvishal** [comment]
Thanks for your response.
Is there a workaround that I can use?

Thanks.

**Comment by kagaya85** [comment]
Yeah, try switch to the branch `feature-serverless` which we are working on, the latest deployment files are available on this branch.
But there may be bugs because we haven't done sufficient testing. 
We will soon merge it to master.😀

**Comment by salehsedghpour** [comment]
I've tried the suggested branch. But the issue still exists.

**Comment by kagaya85** [comment]
Hi @salehsedghpour , we have merged the fix into master, please try using the deployment file on mater directly.

If there is still an issue please feel free to contact us with the relevant information and we will follow up on the issue.

**Comment by salehsedghpour** [comment]
Hi, @kagaya85. I've checked it again, the issue still exists on both **quickstart-k8s** and **k8s-with-istio**. Here is the describe of the pod for  ui:

```
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  2m51s              default-scheduler  Successfully assigned default/ts-ui-dashboard-7b7f64d97c-47mxb to r08
  Normal   Pulling    2m27s              kubelet            Pulling image "docker.io/istio/proxyv2:1.9.2"
  Normal   Pulled     2m5s               kubelet            Successfully pulled image "docker.io/istio/proxyv2:1.9.2" in 21.707965687s
  Normal   Created    116s               kubelet            Created container istio-init
  Normal   Started    98s                kubelet            Started container istio-init
  Normal   Pulled     70s                kubelet            Successfully pulled image "codewisdom/ts-ui-dashboard:0.0.4" in 21.429806269s
  Normal   Pulling    61s                kubelet            Pulling image "docker.io/istio/proxyv2:1.9.2"
  Normal   Pulled     56s                kubelet            Successfully pulled image "docker.io/istio/proxyv2:1.9.2" in 4.912279912s
  Normal   Created    53s                kubelet            Created container istio-proxy
  Normal   Started    52s                kubelet            Started container istio-proxy
  Normal   Pulled     49s                kubelet            Successfully pulled image "codewisdom/ts-ui-dashboard:0.0.4" in 2.338106723s
  Normal   Pulling    31s (x3 over 92s)  kubelet            Pulling image "codewisdom/ts-ui-dashboard:0.0.4"
  Normal   Pulled     29s                kubelet            Successfully pulled image "codewisdom/ts-ui-dashboard:0.0.4" in 1.699852908s
  Normal   Created    27s (x3 over 66s)  kubelet            Created container ts-ui-dashboard
  Normal   Started    26s (x3 over 61s)  kubelet            Started container ts-ui-dashboard
  Warning  BackOff    12s (x5 over 46s)  kubelet            Back-off restarting failed container

```

**Comment by ansnadeem** [comment]
I have the same issue after deploying through docker-compose
`[emerg] 1#1: host not found in upstream "ts-avatar-service"`
And the dashboard container keeps crashing because of it.

**Comment by kagaya85** [comment]
Sorry, can I ask which kubernetes deployment yaml file you are using?

It may caused by the nginx cannot be started because the `ts-avatar-service` is not found, I need to confirm if this is due to a misconfiguration of the deployment file.

**Comment by ansnadeem** [comment]
> Sorry, can I ask which kubernetes deployment yaml file you are using?
> 
> It may caused by the nginx cannot be started because the `ts-avatar-service` is not found, I need to confirm if this is due to a misconfiguration of the deployment file.

In my case, I'm simply using docker-compose up to start all the services, everything starts up, however, whenever I try to hit the dashboard at 8080, the dashboard crashes with the error.

**Comment by kagaya85** [comment]
@ansnadeem Sorry, we forgot to update our docker-compose file and just add the avatar-service in `docker-compose.yml` to start nginx properly, like [this](https://github.com/FudanSELab/train-ticket/pull/153/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R549-R556)
